test/extensions/filters/http/proto_api_scrubber/integration_test.cc

```diff
@@ -0,0 +1,316 @@
+
+  upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, true);
+  ASSERT_TRUE(response->waitForEndStream());
+}
+
+TEST_P(ProtoApiScrubberIntegrationTest, StreamingPassesThrough) {
+  config_helper_.prependFilter(getFilterConfig(apikeysDescriptorPath()));
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+
+  auto req1 = makeCreateApiKeyRequest(R"pb(parent: "req1")pb");
+  auto req2 = makeCreateApiKeyRequest(R"pb(parent: "req2")pb");
+  auto req3 = makeCreateApiKeyRequest(R"pb(parent: "req3")pb");
+
+  Buffer::OwnedImpl combined_request;
+  combined_request.move(*Grpc::Common::serializeToGrpcFrame(req1));
+  combined_request.move(*Grpc::Common::serializeToGrpcFrame(req2));
+  combined_request.move(*Grpc::Common::serializeToGrpcFrame(req3));
+
+  auto request_headers = Http::TestRequestHeaderMapImpl{{":method", "POST"},
+                                                        {":path", kCreateApiKeyMethod},
+                                                        {"content-type", "application/grpc"},
+                                                        {":authority", "host"},
+                                                        {":scheme", "http"}};
+
+  auto response = codec_client_->makeRequestWithBody(request_headers, combined_request.toString());
+  waitForNextUpstreamRequest();
+
+  Buffer::OwnedImpl data;
+  data.add(upstream_request_->body());
+  checkSerializedData<apikeys::CreateApiKeyRequest>(data, {req1, req2, req3});
+
+  upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, true);
+  ASSERT_TRUE(response->waitForEndStream());
+}
+
+// ============================================================================
+// TEST GROUP 2: SCRUBBING LOGIC
+// ============================================================================
+
+TEST_P(ProtoApiScrubberIntegrationTest, ScrubTopLevelField) {
+  config_helper_.prependFilter(
+      getFilterConfig(apikeysDescriptorPath(), kCreateApiKeyMethod, "parent"));
+  initialize();
+
+  auto original_proto = makeCreateApiKeyRequest(R"pb(
+    parent: "sensitive-data"
+    key { display_name: "public" }
+  )pb");
+
+  auto response = sendGrpcRequest(original_proto, kCreateApiKeyMethod);
+  waitForNextUpstreamRequest();
+
+  apikeys::CreateApiKeyRequest expected = original_proto;
+  expected.clear_parent();
+
+  Buffer::OwnedImpl data;
+  data.add(upstream_request_->body());
+  checkSerializedData<apikeys::CreateApiKeyRequest>(data, {expected});
+
+  upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, true);
+  ASSERT_TRUE(response->waitForEndStream());
+}
+
+TEST_P(ProtoApiScrubberIntegrationTest, ScrubNestedField) {
+  config_helper_.prependFilter(
+      getFilterConfig(apikeysDescriptorPath(), kCreateApiKeyMethod, "key.display_name"));
+  initialize();
+
+  auto original_proto = makeCreateApiKeyRequest(R"pb(
```

Comment:
Will it be also useful to test the following cases:
- more than 1 key exists (say `another_name` in addition to `display_name`) and only one (`display_name`) is scrubbed?
- not sure if it's valid but a case where there's a double-nested field with the same name `key { key: { display_name: "sensitive" } }`.

These may already be addressed by the unit tests, and in that case, that should be sufficient.

Reply:
SGTM